Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp

```diff
@@ -462,7 +462,7 @@ ErrorOr<void> initialize_main_thread_vm()
         auto destination = Fetch::Infrastructure::Request::Destination::Script;
 
         // 11. Let fetchClient be settingsObject.
-        Optional<HTML::EnvironmentSettingsObject&> fetch_client = *settings_object;
+        Optional<JS::NonnullGCPtr<HTML::EnvironmentSettingsObject>> fetch_client = *settings_object;
```

Comment:
Here and in LoadRequest+ResourceLoader below - `Optional<JS::NonnullGCPtr<HTML::EnvironmentSettingsObject>>` just sounds like a long-winded way of saying `JS::GCPtr<HTML::EnvironmentSettingsObject>`. But in this instance, can't we just do:
```c++
JS::NonnullGCPtr fetch_client { *settings_object };
```

Reply:
Actually `settings_object` itself is declared as `Optional<HTML::EnvironmentSettingsObject&>` at the top of this function. So if you want to get rid of that non-GCPtr usage:
```c++
JS::NonnullGCPtr settings_object { HTML::current_settings_object() };
```

Then `fetch_client`'s assignment just becomes:
```c++
auto fetch_client = settings_object;
```